lib/JSON.cpp

```diff
@@ -361,6 +361,7 @@ JSONTranslator::DecodeValue(const llvm::json::Object *obj, const char *desc,
 
   if (auto maybe_type_str = obj->getString("type")) {
     std::string spec = maybe_type_str->str();
+    decl.type_string = spec;
```

Comment:
I think you missed doing this assignment in ParseJsonIntoCallableDecl

Reply:
Doesn't it use `DecodeValue` as well?